services/venice-controller/src/main/java/com/linkedin/venice/controller/datarecovery/DataRecoveryManager.java

```diff
@@ -79,15 +78,20 @@ public void initiateDataRecovery(
       String sourceFabric,
       boolean copyAllVersionConfigs,
       Version sourceFabricVersion) {
-    Version dataRecoveryVersion = sourceFabricVersion.cloneVersion();
-    dataRecoveryVersion.setStatus(VersionStatus.STARTED);
     Store store = veniceAdmin.getStore(clusterName, storeName);
     if (store == null) {
       throw new VeniceNoStoreException(storeName, clusterName);
     }
-    dataRecoveryVersion.setNativeReplicationEnabled(store.isNativeReplicationEnabled());
-    dataRecoveryVersion.setNativeReplicationSourceFabric(sourceFabric);
-    dataRecoveryVersion.setDataRecoveryVersionConfig(new DataRecoveryVersionConfigImpl(sourceFabric, false));
+    int srcFabricVersionNumber = sourceFabricVersion.getNumber();
+    if (srcFabricVersionNumber != version) {
+      sourceFabricVersion.setNumber(version);
+      // Update the push job id as a version with same id cannot be added twice.
+      sourceFabricVersion.setPushJobId("data_recovery_" + sourceFabricVersion.getPushJobId());
```

Comment:
When recovering data to the same version, the version is deleted from dest store during prepare_data_recovery. When adding the version again, there is no version with the same push job id in the store. So we don't need to update the push job id.

But when recovering data to a new version, the version with the same push job id still exists in the dest store. We need to update the push job id of the new version. Otherwise, the new version cannot be added:
https://github.com/linkedin/venice/blob/2404d2387a24c048e759b47d990a0ddee842d4be/services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java#L2077-L2078

Reply:
I see, it makes sense. Would be good if you can add a pointer to this function in the above comment line.

// Update the push job id as a version with same id cannot be added twice (@see VeniceHelixAdmin##addSpecificVersion) . 

or something like that.

Reply:
Sure